tests/ttnn/integration_tests/stable_diffusion/test_perf_stable_diffusion.py

```diff
@@ -0,0 +1,207 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import time
+import torch
+import ttnn
+from tqdm.auto import tqdm
+from loguru import logger
+import pytest
+
+from transformers import CLIPTextModel, CLIPTokenizer
+from diffusers import AutoencoderKL, UNet2DConditionModel
+from diffusers import LMSDiscreteScheduler
+
+from models.utility_functions import (
+    skip_for_wormhole_b0,
+    enable_persistent_kernel_cache,
+    disable_persistent_kernel_cache,
+)
+from models.perf.perf_utils import prep_perf_report
+
+from models.experimental.functional_stable_diffusion.tt.ttnn_functional_unet_2d_condition_model import (
+    UNet2DConditionModel as ttnn_UNet2DConditionModel,
+)
+from ttnn.model_preprocessing import preprocess_model_parameters
+from models.experimental.functional_stable_diffusion.custom_preprocessing import custom_preprocessor
+
+
+NUM_INFERENCE_STEPS = 1  # Number of denoising steps
+BATCH_SIZE = 1
+
+
+def constant_prop_time_embeddings(timesteps, sample, time_proj):
+    timesteps = timesteps[None]
+    timesteps = timesteps.expand(sample.shape[0])
+    t_emb = time_proj(timesteps)
+    return t_emb
+
+
+def guide(noise_pred, guidance_scale, t):  # will return latents
+    noise_pred_uncond, noise_pred_text = noise_pred.chunk(2)
+    noise_pred = noise_pred_uncond + guidance_scale * (noise_pred_text - noise_pred_uncond)
+    return noise_pred
+
+
+def latent_expansion(latents, scheduler, t):
+    latent_model_input = torch.cat([latents] * 2)
+    latent_model_input = scheduler.scale_model_input(latent_model_input, timestep=t)
+    return latent_model_input
+
+
+@skip_for_wormhole_b0()
+@pytest.mark.models_performance_bare_metal
+@pytest.mark.models_performance_virtual_machine
+@pytest.mark.parametrize("model_name", ["CompVis/stable-diffusion-v1-4"])
+@pytest.mark.parametrize(
+    "expected_inference_time, expected_compile_time",
+    (
+        (
+            140,
+            140,
+        ),
+    ),
+)
+def test_perf(device, expected_inference_time, expected_compile_time, model_name, reset_seeds):
```

Comment:
Can we please kick this test off from [https://github.com/tenstorrent-metal/tt-metal/blob/main/tests/scripts/run_performance.sh](https://github.com/tenstorrent-metal/tt-metal/blob/main/tests/scripts/run_performance.sh) ?

Reply:
> Can we please kick this test off from https://github.com/tenstorrent-metal/tt-metal/blob/main/tests/scripts/run_performance.sh ?

Yes. It has been updated in run_performance.sh